### PR TITLE
Brushless v3 updates

### PR DESCRIPTION
**Pull request by jedahan**
By manually reverting the webserial changes, I got it to work correctly.

# Known Issues

1. Tests don't pass
2. Brushless option in UI doesn't seem to update the ebb hardware option - that means if you start the server with the wrong `--hardware` option I don't think you can change it while running.

### CODE REVIEW

**Comment by jedahan** [comment]
Tested today - so far the UI looks like its working, but pen up/down are not. Seeing if I can plot from the CLI first

**Comment by jedahan** [comment]
Added new `cli pen <percent>` command to help debug a bit. Added known issues to the comments in the other PR

File: src/__tests__/ebb.test.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
-    oldBinding = NodeSerialPort.Binding;
-    NodeSerialPort.Binding = MockBinding;
+    oldBinding = SerialPort.Binding;
+    SerialPort.Binding = MockBinding;
   });
   afterAll(() => {
-    NodeSerialPort.Binding = oldBinding;
+    SerialPort.Binding = oldBinding;
     MockBinding.reset();
   });
```
Comment (jedahan) [inline src/__tests__/ebb.test.ts:16]: these need to be fixed like in https://github.com/nornagon/saxi/pull/160

Comment (alexrudd2) [reply src/__tests__/ebb.test.ts:16]: Yeah, sadly `node-serialport` upstream is unresponsive.   https://github.com/serialport/node-serialport/issues/2683

**Comment by alexrudd2** [comment]
Regarding the tests, I'm pretty sure that's from: https://github.com/alexrudd2/saxi/commit/abd9591d8850483bedcbb84762f37f9169c37cd0

The code that was removed in the simplification was related to https://github.com/nornagon/saxi/issues/90 and perhaps shouldn't have been deleted?

**Comment by alexrudd2** [comment]
Discussion about WebSerial here: https://github.com/alexrudd2/saxi/issues/89

I've taken some commits and applied them to my `brushless-v3` as fixup commits.  I intend to keep rebasing that branch until the commits are logically grouped, but did the fixup commits to avoid pulling the rug out from under you.

**Comment by jedahan** [comment]
My first goal is to get brushless support working for my machine. Up until yesterday, this was not the case, so I have been pushing all WIP progress in getting that working to my brushless-v3 branch. I could have made this a draft PR, that might have been a better idea.

My current intentions are to fix the brushless ui option, fix the tests, then rebase to a few logical commits.
Each of these commits could be a separate PR against alexrudd2/brushless-v3 if that is preferred, here's what it might look like:

1. Revert WebSerial
2. Upgrade SerialPort
3. Misc cleanups
4. Add 'pen' command

As cool as WebSerial is, my use case doesn't need it, my browser will not support it, and it seems to be broken for the non-webserial use case.

I am okay maintaining a fork until webserial support is either fixed or just separately built a la the other `IS_WEB` stuff. Happy to brainstorm ways to get this working, and if you want to have a videochat to help plan collaborating a bit better, I'm in EST time zone.

**Comment by alexrudd2** [comment]
> My first goal is to get brushless support working for my machine. Up until yesterday, this was not the case, 

OK, I was in a similar situation but got the brushless motor working a while ago (RPi 4).   Congrats!

> so I have been pushing all WIP progress in getting that working to my brushless-v3 branch. I could have made this a draft PR, that might have been a better idea.
> My current intentions are to fix the brushless ui option, fix the tests, then rebase to a few logical commits.

That's basically my plan too.  I try to keep branches around based off `nornagon:main` and submit them with logical commits/PRs.  I've just made you a collaborator here; let's combine everything together and then refactor `brushless-v3` (or even start fresh with v4).

> As cool as WebSerial is, my use case doesn't need it, my browser will not support it, and it seems to be broken for the non-webserial use case.

It's fairly broken but not totally.  I did manage to get brushless + WebSerial working, and used to hold out hope it could be kept around.  But for the time let's revert it and move forward.

**Comment by jedahan** [comment]
closing in favor of trying to make smaller PRs that do one thing

**Pull request closed without merging**